### PR TITLE
CAS-110 bookings CRN search

### PR DESCRIPTION
**Pull request by froddd**
# Context

https://dsdmoj.atlassian.net/browse/CAS-110

This adds CRN search to the bookings search. The layout of the Bookings search screens is updated to accommodate the new search functionality.

# Changes in this PR

- Add search box to search by CRN
- Clicking 'Clear' clears the search and loads all results
- Move side navigation to horizontal tab bar
- Use full width for search results

### Before

Screenshot 2024-02-29 at 15 50 49

### After

Screenshot 2024-02-29 at 15 50 09

# Release checklist

[Release process documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work? Yes: https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/1545
- [ ] Have they been released to production already? No

## Post-merge checklist

- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -3,5 +3,14 @@
   "singleQuote": true,
   "printWidth": 120,
   "semi": false,
-  "arrowParens": "avoid"
+  "arrowParens": "avoid",
+  "plugins": ["prettier-plugin-jinja-template"],
```
Comment (richpjames) [inline .prettierrc:7]: wise move! I'm stealing this for AP

File: server/views/temporary-accommodation/booking-search/results.njk [hunk 2]
```diff
@@ -19,6 +21,34 @@
     })
   }}
 
+  <div class="govuk-template govuk-!-padding-8 govuk-!-padding-bottom-static-3 govuk-!-margin-bottom-8">
+    <form action="" method="get">
```
Comment (richpjames) [inline server/views/temporary-accommodation/booking-search/results.njk]: should this `form` have an `action`?

**Review by richpjames — DISMISSED** [review]
Thanks for breaking these changes down into easy-to-read commits. It makes it so much easier for me to review.

File: server/views/temporary-accommodation/booking-search/results.njk (hunk 2 above)
Comment (froddd) [reply server/views/temporary-accommodation/booking-search/results.njk]: Added in subsequent commit!

**Review by richpjames — APPROVED** [review]

**Pull request merged**